web/components/NavBarLoggedInDesktop.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactElement } from "react";
+import { NavDefaults } from "@/display-content/NavDefaults";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { Icon } from "@/components/njwds/Icon";
+import { AuthButton } from "@/components/AuthButton";
+import Link from "next/link";
+
+export const NavBarLoggedInDesktop = (): ReactElement => {
+  const { userData } = useUserData();
+
+  const getUser = (): string => {
+    return userData?.user.name
+      ? userData?.user.name
+      : userData?.user.email
+      ? userData?.user.email
+      : NavDefaults.userNotSetText;
+  };
```

Comment:
A few notes here:
- I wonder if this would be more readable as a simple if - else if - else instead of a double ternary 
- Same as for the file below (maybe both should make use of a helper method?)
- This is a type of logic I'd write a test for (uses name if exists, email if not exists, and empty text of userData itself is undefined)

Reply:
Created test, and created helper method.